### ISSUE TITLE
[Question]How to set up a worker per each spec file and run the tests in each spec file order by order?

### ISSUE THREAD

**Issue by YusufOzturk-Development** [microsoft/playwright#19966]
Projects structure 
- a.test.spec
- b.test.spec
- c.test.spec
- d.spec.ts

Each spec file has about 20 tests . When I run npx playwright test script, I want to set up a worker per each spec file and run the tests in each spec file order by order.Thus , the four spec files work parallel  and each test will work order by order in the spec files 
Does default mode supports this logic or should I use different method? I read parallesim and sharding document but I didnt find solution . Maybe I missed something. If do u have an advice, I want to learn

**Comment by aslushnikov** [comment]
@YusufOzturk-Development the mode you describe is the default mode:
- it's described in our [documentation](https://playwright.dev/docs/next/test-parallel):
    > By default, test files are run in parallel. **Tests in a single file are run in order, in the same worker process.**
- we also re-iterate on it in [this video](https://www.youtube.com/watch?v=6RwzsDeEj7Y&list=PLQ6Buerc008fTzmUIPkPo3Eis0vjx7ykn&index=15&t=83s)

So things should work as expected. If they don't please don't hesitate to submit another issue with a demo project that sets up a few spec files and demonstrates unexpected execution order!
